src/protagonist/DLCS.Repository/Spaces/SpaceRepository.cs

```diff
@@ -116,7 +121,7 @@ private async Task<int> GetIdForNewSpace(int requestCustomer)
         CancellationToken cancellationToken, string? name = null,
         bool withApproximateImageCount = false, bool noCache = false)
     {
-        var key = $"space:{customerId}/{spaceId}";
+        var key = $"space:{customerId}/{name ?? spaceId.ToString()}";

```

Comment:
We don't add any restrictions on space names so if customer 123 creates 2 spaces: 

```json
[
  { "Id": 1, "Name": "2" }
  { "Id": 2, "Name": "1" }
]
```

Those could both be legitimately cached as either `space:123:1` or `space:123:2` depending on how the request was made.

Reply:
to avoid this, I could change the key based on the cache type? i.e.: `space:123:name:2` or `space:123:id:1`